### PR TITLE
Wizard recipe: re2c-v3.0.0

### PR DESCRIPTION
**Pull request by lucifer1004**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: re2c
* Version: v3.0.0

@staticfloat please review and merge.

### CODE REVIEW

File: R/re2c/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "re2c"
+version = v"3.0.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/skvadrik/re2c/releases/download/3.0/re2c-3.0.tar.xz", "b3babbbb1461e13fe22c630a40c43885efcfbbbb585830c6f4c0d791cf82ba0b")
```
Comment (giordano) [inline R/re2c/build_tarballs.jl:10]:
```suggestion
version_string = "3.0"
version = VersionNumber(version_string)

# Collection of sources required to complete build
sources = [
    ArchiveSource("https://github.com/skvadrik/re2c/releases/download/$(version_string)/re2c-$(version_string).tar.xz", "b3babbbb1461e13fe22c630a40c43885efcfbbbb585830c6f4c0d791cf82ba0b")
```

File: R/re2c/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/re2c-3.0/
```
Comment (giordano) [inline R/re2c/build_tarballs.jl:15]:
```suggestion
cd $WORKSPACE/srcdir/re2c-*
```

File: R/re2c/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("re2go", :re2go),
+    ExecutableProduct("re2rust", :re2rust),
+    ExecutableProduct("re2c", :re2c)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
```
Comment (giordano) [inline R/re2c/build_tarballs.jl:35]: Why is this needed?

Comment (lucifer1004) [reply R/re2c/build_tarballs.jl]: Removed. Could you explain to me a bit what this package does? I added it here because there were a few times you suggested adding this package. But I did not know its function here.

Comment (giordano) [reply R/re2c/build_tarballs.jl]: It's a package which provides some compiler-specific libraries.  Whenever I tell you to add it it's because there are some dozens of warnings in the logs saying so, that's how I know it's needed, you too can read the log and see if the auditor is telling you to add it :slightly_smiling_face:  Now you removed the package and the auditor isn't saying anything, so it wasn't needed.

Comment (lucifer1004) [reply R/re2c/build_tarballs.jl]: Thanks!
